ruby/stdin.rb

```diff
@@ -0,0 +1,13 @@
+# Test script to get input, transform, and write to stdout
+
+i = 1
+
+while user_input = gets
+  begin
+    user_input.chomp
+    puts "#{i} #{user_input.upcase}"
+    i += 1
+  rescue NoMethodError
```

Comment:
- The loop will exit when `gets` returns `nil` so we don't need to `break`

- `user_input` will never be `nil` inside the loop. Everything is an expression in Ruby, even assignment, so `user_input = gets` will evaluate to whatever `gets` returns (a string or `nil`). And so, the condition of the `while` loop will be a string (which is always truthy) or `nil` (which is falsy). 

I'm not sure sure if I explained that well. I this case it's like the walrus operator in python (e.g. `while (user_input := gets()):`

Reply:
Ok I get it. That makes sense now. Taking out the rescue/break lines.